test/sql/test_iceberg/T/test_iceberg_in_list_predicate

```diff
@@ -0,0 +1,28 @@
+-- name: test_iceberg_in_list_predicate
+create external catalog ice_cat_${uuid0}
+properties
+(
+    "type" = "iceberg",
+    "iceberg.catalog.type" = "hive",
+    "hive.metastore.uris" = "${iceberg_catalog_hive_metastore_uris}"
+);
+set catalog ice_cat_${uuid0};
+create database ice_db_${uuid0};
+use ice_db_${uuid0};
+create table ice_tbl_${uuid0} (
+  col_str string,
+  col_int int
+);
+insert into ice_tbl_${uuid0} values ('1d8cf2a2c0e14fa89d8117792be6eb6f', 2000),
+  ('3e82e36e56718dc4abc1168d21ec91ab', 2000),
+  ('abc', 2000),
+  ('ab1d8cf2a2c0e14fa89d8117792be6eb6f', 2001),
+  ('3e82e36e56718dc4abc1168d21ec91ab', 2001),
+  ('abc', 2001);
+select col_str from ice_tbl_${uuid0} where col_int=2000 and col_str not in (md5('1d8cf2a2c0e14fa89d8117792be6eb6f'), 'abc');
```

Comment:
Better to introduce some tests with NULL values

Reply:
done